plugins/by-name/web-devicons/default.nix

```diff
@@ -14,4 +14,36 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
     color_icons = true;
     strict = true;
   };
+
+  extraOptions = {
+    customIcons = defaultNullOpts.mkAttrsOf (lib.types.submodule {
+      freeformType = lib.types.anything;
+
+      options = {
+        icon = defaultNullOpts.mkStr null "Icon to use.";
+        color = defaultNullOpts.mkStr null "Color of the icon.";
+        cterm_color = defaultNullOpts.mkStr null "Cterm color of the icon.";
+        name = defaultNullOpts.mkStr null "Name to replace with icon.";
+      };
+    }) { } ''Custom overrides for icons.'';
+
+    defaultIcon = mkNullOrOption (lib.types.submodule {
+      options = {
+        icon = defaultNullOpts.mkStr null "Icon to use.";
+        color = defaultNullOpts.mkStr null "Color of the icon.";
+        cterm_color = defaultNullOpts.mkStr null "Cterm color of the icon.";
+      };
+    }) ''Set the default icon when none is found.'';
+  };
+
+  extraConfig = cfg: {
+    plugins.web-devicons.luaConfig.content =
+      lib.optionalString (cfg.customIcons != null) ''
+        require('nvim-web-devicons').set_icon(${toLuaObject cfg.customIcons})
+      ''
+      + lib.optionalString (cfg.defaultIcon != null) ''
+        require('nvim-web-devicons').set_default_icon(
+          ${toLuaObject cfg.defaultIcon.icon}, ${toLuaObject cfg.defaultIcon.color}, ${toLuaObject cfg.defaultIcon.cterm_color})
+      '';
```

Comment:
Does it matter if this happens before/after `setup`? Currently the merge order will be determined by the definition order in mkNeovimPlugin's `mkMerge` call.

Reply:
Can't find anything about order mattering, but might be safer to assume post.